scripts/packageYoutubedown.js

```diff
@@ -0,0 +1,149 @@
+
+const childProcess = require('child_process')
+const commander = require('commander')
+const fs = require('fs-extra')
+const mkdirp = require('mkdirp')
+const path = require('path')
+const replace = require('replace-in-file')
+const request = require('request')
+const util = require('../lib/util')
+const ntpUtil = require('../lib/ntpUtil')
+
+const getOutPath = (outputFilename) => {
+  let outPath = path.join('build')
+  if (!fs.existsSync(outPath)) {
+    fs.mkdirSync(outPath)
+  }
+  outPath = path.join(outPath, 'youtubedown')
+  if (!fs.existsSync(outPath)) {
+    fs.mkdirSync(outPath)
+  }
+  return path.join(outPath, outputFilename)
+}
+
+const stageFiles = (version, outputDir) => {
+  const scriptFile = getOutPath('youtubedown.js')
+  const outputScriptFile = path.join(outputDir, 'youtubedown.js')
+  console.log('copy ', scriptFile, ' to:', outputScriptFile)
+  fs.copyFileSync(scriptFile, outputScriptFile)
+
+  // Fix up the manifest version
+  const originalManifest = getOriginalManifest()
+  const outputManifest = path.join(outputDir, 'manifest.json')
+  console.log('copy manifest file: ', originalManifest, ' to: ', outputManifest)
+  const replaceOptions = {
+    files: outputManifest,
+    from: /0\.0\.0/,
+    to: version
+  }
+  fs.copyFileSync(originalManifest, outputManifest)
+  replace.sync(replaceOptions)
+}
+
+const generateManifestFile = (publicKey) => {
+  const manifestFile = getOriginalManifest()
+  const manifestContent = {
+    description: 'Brave wrapper for youtubedown (jwz.org)',
+    key: publicKey,
+    manifest_version: 2,
+    name: 'youtubedown',
+    version: '0.0.0'
+  }
+  fs.writeFileSync(manifestFile, JSON.stringify(manifestContent))
+}
+
+function downloadFileSync (downloadUrl, destPath) {
+  return new Promise(function (resolve, reject) {
+    // Not sure why below code doesn't work. callback is not called.
+    // request(downloadUrl, async function (error, response, body) {
+    //   if (error) {
+    //     console.error(`Error trying to download ${downloadUrl}:`, error)
+    //     return reject(error)
+    //   }
+
+    //   if (response && response.statusCode === 200) {
+    //     fs.writeFileSync(destPath, body)
+    //     resolve()
+    //   }
+
+    //   const errorText = response
+    //     ? `Invalid response code: ${response.statusCode}:`
+    //     : 'Response was null or empty'
+    //   console.error(errorText)
+    //   return reject(errorText)
+    // })
+
+    request
+      .get(downloadUrl)
+      .on('response', function(response) {
+        resolve()
+      })
+      .on('error' , function(error) {
+        return reject()
+      })
+      .pipe(fs.createWriteStream(destPath))
+  })
+}
+
+async function downloadLatestYoutubedown (bucketUrl) {
+  // TODO(bsclifton): doesn't handle errors (ex: 403)
+  // script is currently getting a 403 when fetched :(
+  // Because of 403 from 'https://www.jwz.org/hacks/youtubedown.js', we copied
+  // youtubedown.js to s3 bucket.
```

Comment:
@bsclifton As you wrote, I also can't get script from jwz.org.
As a workaround, bucket hosts it instead.

Reply:
I think we can supply a user-agent and it should download just fine. Maybe we can try a few things with `curl` - it reproduces easily with that

Reply:
See https://bravesoftware.slack.com/archives/CKJCXTP9B/p1596702301074300

Reply:
@bsclifton  Works perfect with UA. Thanks!